src/compiler/scanner.ts

```diff
@@ -1226,17 +1243,60 @@ namespace ts {
                 case CharacterCodes.doubleQuote:
                     return "\"";
                 case CharacterCodes.u:
+                    if (isTaggedTemplate) {
+                        // '\u'
+                        if (pos < end && !isHexDigit(text.charCodeAt(pos)) && text.charCodeAt(pos) !== CharacterCodes.openBrace) {
+                            tokenFlags |= TokenFlags.ContainsInvalidEscape;
+                            return text.substring(start, pos);
+                        }
+
+                        // '\u0' or '\u00' or '\u000'
+                        for (let i = 0; i < 3; i++) {
+                            if (pos + i + 1 < end && isHexDigit(text.charCodeAt(pos + i)) &&  !isHexDigit(text.charCodeAt(pos + i + 1)) && text.charCodeAt(pos + i + 1) !== CharacterCodes.openBrace) {

```

Comment:
Remove double space here

Reply:
Can you fold the previous `if` into part of the loop and iterate over 4 times? If it gets too gross, don't worry about it